### PR TITLE
Update the Material 3 theme in the demo app

### LINKED ISSUE #2188: Demo app reskinning.
**Describe the bug**
Update visual appearance of demo app.
Refer below link for reference
https://github.com/google/fhir-app-examples/tree/main/configurablecare

**Component**
reference app

**To Reproduce**

**Expected behavior**

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: [e.g. Pixel4a emulator]
 - Android version: [e.g. Settings -> About phone -> Android version]
 - Build number: [e.g. Settings -> About phone -> Build number]

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes

**Description**
 #2188
In the context of the demo app reskinning, we are now extending support for the Material Design 3 (M3) theme in the demo app. This pull request (PR) primarily involves transitioning from the Material Design 2 (M2) theme to the Material Design 3 theme. Additionally, we are updating the M3 theme-based color schemes to align with the primary color. If necessary, the color palette can be further adjusted without affecting the existing implementation.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
Screenshot 2023-09-27 at 3 14 01 PM

File: demo/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -28,6 +28,94 @@
     <color name="unknown_risk_background">#10156125</color>
     <color name="unknown_risk">#ffffff</color>
     <color name="syncing_background">#E8F0FE</color>
-    <color name="white">#FFFFFFFF</color>
     <color name="disable_state">#9AFFFFFF</color>
+
```
Comment (jingtang10) [inline demo/src/main/res/values/colors.xml:32]: i think colorPrimary on line 18 of this flie should definitely be removed. some others too. please take a look

File: demo/src/main/res/values-night/themes.xml
Comment (jingtang10) [inline demo/src/main/res/values-night/themes.xml:1]: rename to styles.xml

**Review by jingtang10 — APPROVED** [review]

**Comment by santosh-pingle** [comment]
Screenshot 2023-09-28 at 4 56 39 PM
Screenshot 2023-09-28 at 4 56 52 PM

File: demo/src/main/res/values/colors.xml (hunk 1 above)
Comment (santosh-pingle) [reply demo/src/main/res/values/colors.xml:28]: done.

File: demo/src/main/res/values-night/themes.xml
Comment (santosh-pingle) [reply demo/src/main/res/values-night/themes.xml:1]: done.

**Comment by santosh-pingle** [comment]
Configuration plan : 
Screenshot 2023-09-29 at 10 40 16 AM
Screenshot 2023-09-29 at 10 40 26 AM

Screenshot 2023-09-29 at 10 40 45 AM
Screenshot 2023-09-29 at 10 40 57 AM
Screenshot 2023-09-29 at 10 41 06 AM

File: demo/src/main/res/values/colors.xml [hunk 2]
```diff
@@ -28,6 +24,67 @@
...
+    <color name="onSecondaryContainer_blue_10">#001D35</color>
+
+    <color name="tertiary_green_40">#146C2E</color>
+    <color name="onTertiary_green_100">#FFFFFF</color>
+    <color name="tertiaryContainer_green_90">#C4EED0</color>
+    <color name="onTertiaryContainer_green_10">#072711</color>
+
+    <color name="error_red_40">#B3261E</color>
+    <color name="errorContainer_red_100">#FFFFFF</color>
+    <color name="onError_red_90">#F9DEDC</color>
```
Comment (jingtang10) [inline demo/src/main/res/values/colors.xml:46]: swap?

**Review by jingtang10 — APPROVED** [review]

File: demo/src/main/res/values/colors.xml (hunk 2 above)
Comment (santosh-pingle) [reply demo/src/main/res/values/colors.xml:46]: done.

**Pull request merged**